src/helpers/buffers/send.rs

```diff
@@ -157,21 +136,21 @@ impl Config {
 
 #[cfg(test)]
 mod tests {
-    use crate::helpers::buffers::send::{Config, PushError};
+    use crate::helpers::buffers::send::{ByteBuf, Config, PushError};
     use crate::helpers::buffers::SendBuffer;
     use crate::helpers::Role;
     use crate::protocol::{RecordId, UniqueStepId};
-    use rand::seq::SliceRandom;
-    use rand::thread_rng;
-    use std::cmp::Ordering;
+
+    use tinyvec::array_vec;
 
     use crate::helpers::fabric::{ChannelId, MessageEnvelope};
 
     impl Clone for MessageEnvelope {
```

Comment:
could you not just `#[derive(Clone, Copy)]` for `MessageEnvelope`?

Reply:
I did not want to do that to avoid making it cloneable by accident. It it a mistake to try to clone it. On the other hand, it is convenient to clone it in these unit tests. Note that I could've done this `#[cfg_attr(test, Clone)]`  instead. For now it is only used in one place and maybe it will be gone soon